src/transcoding/transcode/stream.c

```diff
@@ -56,7 +56,7 @@ static int
 tvh_stream_setup(TVHStream *self, TVHCodecProfile *profile, tvh_ssc_t *ssc)
 {
     enum AVCodecID icodec_id = streaming_component_type2codec_id(ssc->es_type);
-    AVCodec *icodec = NULL, *ocodec = NULL;
+    const AVCodec *icodec = NULL, *ocodec = NULL;
```

Comment:
Why would you create a const null-pointer? It's modified later on aswell, so const is not correct.

Reply:
When I tried to upgrade to ffmpeg5, instances with 'AVCodec' were flagged .... we can wait until I fix all other issues and check if works without this line. All instances (initialized or not with null) of AVCodec are modified later on .... What I can tell you is that transcoding works with this change. We can remove the initialization will null ... I don't think it matters. Do you want me to check that?

Reply:
It doesn't make sense like this. I assume what you want is to cast it as `const` when passing it to ffmpeg. I'm not sure why the compiler isn't complaining about assigning something to a const variable... I'll need to understand what's going on, then I can tell you what the best way is.

Reply:
I searched through ffmpeg code (examples) and is always defined without initialization (to null). Let me remove those '= NULL' and see if are working without those. I don't think it makes a difference ... :) but let's fixed the 'right' way.

Reply:
It doesn't work without explicit initialization to NULL :( .
![init](https://user-images.githubusercontent.com/4031821/227099671-8752f508-a1f4-4f30-8e5b-870f8d96bf67.png)

Reply:
The problem is that `const` means constant, but you are modifying the constant. I don't get why the compiler isn't loudly complaining about const variables being changed, I might have to ask someone else to explain this to me.

That null assignment isn't a problem, the problem is that a const variable should never change, that's why it's const.

Reply:
If you prefer to lower the risk, I can wrap all changed for ffmpeg5 behind '--enable-ffmpeg5' (that will be default no). This way we can get more feedback for other platforms.

Reply:
`const AVCodec *icodec` is a mutable pointer to immutable `const AVCodec`
  while
`AVCodec *const icodec` is an immutable pointer to mutable `AVCodec`

Can be confusing especially when coding in many languages :)